### PR TITLE
feat:added the animal class and tests

### PR DESCRIPTION
**Pull request by HeirJordan3**
I think I might have this in the wrong place but trying to see if I am on the right track so I am able to help with the students.

### CODE REVIEW

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  public void setColor(String color) {
+    this.color = color;
+  }
+
+  public void setSound(String sound) {
+    this.sound = sound;
+  }
+
+  public class InvalidAgeException extends Exception {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:89]: Should use the exception in one of your methods.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  public void changeColor(String newColor) {
+    this.color = newColor;
+    System.out.println("The " + this.name + " has changed color to " + this.color + ".");
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public boolean isCanBreathe() {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:53]: Should just be `getCanBreathe`.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    BIRD,
+    REPTILE,
+    FISH,
+    AMPHIBIAN,
+    INSECT
+  }
+
+  private String name;
+  private boolean canBreathe;
+  private int age;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:16]: Prefer to use java.time.Instant to set a birth date and then calculate the age from that.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        + ", color="
+        + color
+        + ", sound="
+        + sound
+        + "]";
+  }
+
+  public Object getType() {
+    if (this.type == null) {
+      throw new UnsupportedOperationException("Unimplemented method 'getType'");
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:117]: Firstly, I don't think type will ever be null so this line will never be reached. Also, the exception message doesn't make sense. How about `Type is not set for this instance`?

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.codedifferently.lesson7.Jordan_Eldridge.Animal.AnimalType;
+import org.junit.jupiter.api.Test;
+
+public class AnimalTest {
+
+  @Test
+  public void testCanBreathe() {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:11]: The first four tests are really just testing the same thing (that the animal has been constructed correctly once the constructor is called). Merge into one that just asserts all the properties are set correctly.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
+package com.codedifferently.lesson7.Jordan_Eldridge;
+
+public class Animal {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:3]: Missing a collection and conditions.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package com.codedifferently.lesson7.Jordan_Eldridge;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:1]: Prefer package name that is lowercase alphanumeric, update the file path accordingly.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java (hunk 1 above)
Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:89]: yup makes sense sorry about that !

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java (hunk 6 above)
Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:3]: I did not see that in the GitHub requirements is that something mentioned during the lecture ?

Comment (anthonydmays) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:3]: Boo, I totally forgot to add that. Nevermind, please ignore.

Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:3]: you the man !

Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:3]: this assignment is good though alot of them are struggling with tests!

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java [hunk 8]
```diff
@@ -1,40 +1,69 @@
...
+    assertEquals("This animal can fly!", animal.canFly());
   }
 
   @Test
-  public void testGetColor() {
-    Animal animal = new Animal("Dog", true, 5, "Brown", "Bark", AnimalType.MAMMAL);
-    assertEquals("Brown", animal.getColor());
+  public void testMakeSound() {
+    Animal animal = new Animal("lizard", true, 3, "Green", "Hissing", AnimalType.REPTILE);
+    animal.makeSound();
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:31]: In order for this to be a valid test, there needs to be a proper assertion that will fail if the code changes the expected behavior.

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java [hunk 9]
```diff
@@ -1,40 +1,69 @@
...
   }
 
   @Test
-  public void testSetName() {
+  public void testSetAge() {
     Animal animal = new Animal("Dog", true, 5, "Brown", "Bark", AnimalType.MAMMAL);
-    animal.setName("Cat");
-    assertEquals("Cat", animal.getName());
+    Exception exception =
+        assertThrows(
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:44]: nit: You can also use `assertThatThrowsBy` to check assertions, which is how the cool kids do it nowadays I believe.

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java [hunk 10]
```diff
@@ -1,40 +1,69 @@
...
+            });
+
+    String expectedMessage = "Age cannot be negative.";
+    String actualMessage = exception.getMessage();
+
+    assertTrue(actualMessage.contains(expectedMessage));
+  }
+
+  @Test
+  public void testGetAllAnimals() {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:57]: Nice!

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java [hunk 11]
```diff
@@ -1,4 +1,7 @@
-package com.codedifferently.lesson7.Jordan_Eldridge;
+package com.codedifferently.lesson7.jordan_eldridge;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:1]: Folder names should also be lowercase to match the package name, and you don't need the underscore :D

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java (hunk 11 above)
Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/Jordan_Eldridge/Animal.java:1]: ok sweet learning alot by doing this thank you !

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java (hunk 8 above)
Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:31]: gotcha think I made the fix.

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java (hunk 9 above)
Comment (HeirJordan3) [reply lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:44]: I could not find out where this was at !

Comment (anthonydmays) [reply lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/Jordan_Eldridge/AnimalTest.java:44]: I misspelled it (should be `assertThatThrownBy`), but a Google search leads me to this page. You should be able to import and use it. You can also see example usage in PR #203. 

https://www.javadoc.io/doc/org.assertj/assertj-core/3.8.0/org/assertj/core/api/Assertions.html

**Review by anthonydmays — APPROVED** [review]
LGTM

**Review by anthonydmays — CHANGES_REQUESTED** [review]
Sorry, approved too early

**Review by HeirJordan3 — COMMENTED** [review]
i just made one more push not sure if the approved to early was for the most recent one.

**Comment by anthonydmays** [comment]
> i just made one more push not sure if the approved to early was for the most recent one.

Yup, I approved to early. You also need to sync your branch, it's out of date.

**Pull request closed without merging**